cmake/public/LoadHIP.cmake

```diff
@@ -299,6 +299,8 @@ if(HIP_FOUND)
     message("ROCm version < 4.1; disablng asserts")
   endif()
 
+  enable_language(HIP)
+
```

Comment:
My expectation was this this whole file would be rendered unnecessary due to this PR: https://github.com/pytorch/pytorch/pull/80469?

Reply:
above PR [80469](https://github.com/pytorch/pytorch/pull/80469) installs LoadHIP.cmake so that it is visible to any other project trying to use it. 
Above change is to info cmake to accept HIP code. This change is NOT to related tensorpipe enablement, but a change good to have.

Reply:
Does the `enable_language(HIP)` allow us to use some additional feature of HIP? Wondering why we don't need/have it in the pytorch LoadHIP.cmake.

Also, should we have a different PR to remove the LoadHIP.cmake file from tensorpipe so we can use the PyTorch file?

Reply:
Sorry, I misunderstood that this is the pytorch file itself, not in tensorpipe repo. *facepalm* ..... please disregard my comments about removing this file. As discussed offline, let's just add a comment for the `enable_language(HIP)` line to provide some more context of why we added this.